plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerConnectorTest.java

```diff
@@ -351,9 +351,9 @@ public void testPredicatePushdown()
             assertThat(query(joinPushdownEnabled, "SELECT c.name, n.name FROM customer c JOIN nation n ON c.custkey = n.nationkey WHERE n.name = 'POLAND'"))
                     .isFullyPushedDown();
 
-            // join on varchar columns is not pushed down
-            assertThat(query(joinPushdownEnabled, "SELECT c.name, n.name FROM customer c JOIN nation n ON c.address = n.name"))
-                    .joinIsNotFullyPushedDown();
+            // join on varchar columns
+            assertThat(query(joinPushdownEnabled, "SELECT n.name, n2.regionkey FROM nation n JOIN nation n2 ON n.name = n2.name"))
+                    .isFullyPushedDown();
```

Comment:
Could you please test this on column which contain `null` too?

Reply:
Looks like by default for SQLServer columns are nullable, if they not participate in primary keys e.t.c.
https://social.msdn.microsoft.com/Forums/sqlserver/en-US/16c39868-96b1-457c-a3a5-f291ba75d635/whether-a-column-is-nullable-by-default?forum=transactsql

Reply:
I wondered if we have a test for the `SqlServerClient#getCaseSensitivityForColumns` if it works correctly for the columns which actually have nulls.
```
ResultSetMetaData metadata = preparedStatement.getMetaData();
metadata.isCaseSensitive(column)
```
But it does not relate to exactly this PR, but to previous effort.

Reply:
Yes, I think for this PR it doesn't make sense for JOINS.
But for previous one where we enabled PREDICATE pushdowns it make sense to enhance it with some test cases like:
```
       assertThat(query("SELECT regionkey, nationkey, name FROM nation WHERE name IS NULL"))
                .isFullyPushedDown();
```

However we have support for NULLable pushdowns:
```
        if (domain.isNullableSingleValue()) {
            return FULL_PUSHDOWN.apply(session, domain);
        }
```

But it worth to such test cases, similar to what we have in Postgress:

```
    @Test
    public void testIsNullPredicatePushdown()
```